client/src/modules/email-report/modal/categories-people-list-ctrl.js

```diff
@@ -0,0 +1,126 @@
+angular.module('bhima.controllers')
+  .controller('CategoriesPeopleListController', CategoriesPeopleListController);
+
+CategoriesPeopleListController.$inject = [
+  '$state', 'EmailReportService', 'CategoriesPeopleService', 'SessionService', 'util',
+  'NotifyService', 'ScrollService', 'bhConstants', 'uiGridConstants',
+];
+
+
+/**
+ * Report group Controller
+ *
+ * @description
+ *CategoriesPeopleListController give the possibility to select a group of people
+ * from de database, they can be users, patients, donators..
+ * Once this list is displayed in a table , the user can select one of theme
+ * for eache person the application can get automaticly his email and name 
+ * 
+ * This selection is usefull if the use don't know a person email or name but he is registered
+ * in the database
+ */
+
+function CategoriesPeopleListController($state, EmailReportSvc, CategoriesPeople, Session, util, Notify, ScrollTo, bhConstants, uiGridConstants) {
+
+  var vm = this;
+  //emailReport is the object that contains : name,email, frequence and report-group
+  vm.listCategoriesPeople = CategoriesPeople.list;
+  vm.selectedCategory = {};
+
+  vm.loading = false;
+  vm.hasError = false;
+
+  function load(filters) {
+    /*
+     category people index is save in the localStorage as expained after
+     if there's no index, we can use it, in the most case when the user refresh the page
+     */
+    if (!$state.params.index_current_category) {
+      $state.params.index_current_category = localStorage.getItem('index_current_category');
```

Comment:
If you want to see the implementation of AppCache, you can find it in our services: [AppCache.js](https://github.com/IMA-WorldHealth/bhima-2.X/blob/master/client/src/js/services/AppCache.js).